extensions/GitExtension/FileExplorerGitIntegration/Models/GitConfiguration.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DevHome.Common.Helpers;
+using DevHome.Common.Services;
+using Serilog;
+using Windows.Storage;
+
+namespace FileExplorerGitIntegration.Models;
+
+public class GitConfiguration : IDisposable
+{
+    public GitExecutableConfigOptions GitExecutableConfigOptions { get; set; }
+
+    private readonly FileService fileService;
+
+    private string GitExeInstallPath { get; set; } = string.Empty;
+
+    private readonly object fileLock = new();
+
+    private readonly ILogger log = Log.ForContext<GitDetect>();
+
+    private readonly string tempConfigurationFileName = "TemporaryGitConfiguration.json";
+
+    public GitConfiguration(string? path)
+    {
+        if (RuntimeHelper.IsMSIX)
+        {
+            GitExecutableConfigOptions = new GitExecutableConfigOptions
```

Comment:
NIT: The difference between these clauses is a string assignment.  this can be changed to assigning a string, then making the GitExecutableConfigOptions.

Reply:
I agree, much cleaner.